### PR TITLE
Fix mem generic rpmb

### PR DESCRIPTION
**Pull request by litchipi**
Same purpose as #1 , but changes the generic rpmb driver instead of the optee-only code.

### CODE REVIEW

**Comment by litchipi** [comment]
Tests OK using this patch.
As the mmc driver expect alignment checks BEFORE calling, an alignment check inside the mmc driver would be overlapping.
This solution would then be preferred.

**Comment by litchipi** [comment]
Squashed the patches into a single commit, merged the commit message from previous fix into new patch and adapt its message so it fits the new context on why this change is necessary (more "globally" than before).

**Comment by litchipi** [comment]
indent fix + typo fix after patman check

File: drivers/mmc/rpmb.c [hunk 1]
```diff
@@ -480,10 +480,24 @@ int mmc_rpmb_route_frames(struct mmc *mmc, void *req, unsigned long reqlen,
...
+	void *rpmb_data = NULL;
+	int ret;
 
 	if (reqlen % sizeof(struct s_rpmb) || rsplen % sizeof(struct s_rpmb))
 		return -EINVAL;
 
-	return rpmb_route_frames(mmc, req, reqlen / sizeof(struct s_rpmb),
-				 rsp, rsplen / sizeof(struct s_rpmb));
+	if (!IS_ALIGNED((uintptr_t)req, ARCH_DMA_MINALIGN)) {
+		/* memory alignment is required by MMC driver */
```
Comment (etienne-lms) [inline drivers/mmc/rpmb.c]: s/memory/Memory/  (for consistency in this source file)

**Review by etienne-lms — COMMENTED** [review]
Minor comments only.
In commit log, i would rephrase the 1st sentence:
```patch
-The MMC driver expect alignment check to be done prior the call, and
-this particular API lacks this checking process.
+Fix mmc_rpmb_route_frames() implementation to comply with most
+MMC drivers that expect some alignment of MMC data frames in memory.
```

**Comment by etienne-lms** [comment]
otherwise looks all good to me.

**Comment by etienne-lms** [comment]
+1
